ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -32,107 +36,260 @@ public class GltfModelExporter
+                    for (var i = 0; i < meshes.Length; i++)
+                    {
+                        var node = AddMeshNode(exportedModel, scene, Model,
+                            meshes[i].Name, meshes[i].Mesh, Model.GetSkeleton(i));
+
+                        if (node == null) continue;
+                        // Swap Rotate upright, scale inches to meters.
+                        node.WorldMatrix = Transform * TRANSFORMSOURCETOGLTF;
+                    }
                 }
+            }
 
-                var hasJoints = skeleton.AnimationTextureSize > 0;
-                var exportedMesh = CreateGltfMesh(name, mesh, exportedModel, hasJoints);
-                var hasVertexJoints = exportedMesh.Primitives.All(primitive => primitive.GetVertexAccessor("JOINTS_0") != null);
+            // Then the Entities
+            foreach (var lumpName in world.GetEntityLumpNames())
+            {
+                if (lumpName == null) continue;
+                var entityLumpResource = FileLoader.LoadFile(lumpName + "_c");
+                if (entityLumpResource == null) continue;
 
-                if (hasJoints && hasVertexJoints)
-                {
-                    var skeletonNode = scene.CreateNode(name);
-                    var joints = CreateGltfSkeleton(skeleton, skeletonNode);
+                var entityLump = (VEntityLump)entityLumpResource.DataBlock;
 
-                    scene.CreateNode(name)
-                        .WithSkinnedMesh(exportedMesh, Matrix4x4.Identity, joints);
+                LoadEntityMeshes(exportedModel, scene, entityLump);
+            }
 
-                    // Rotate upright, scale inches to meters.
-                    skeletonNode.WorldMatrix = TRANSFORMSOURCETOGLTF;
+            WriteModelFile(exportedModel, fileName);
+        }
 
-                    // Add animations
-                    var animations = GetAllAnimations(model);
-                    foreach (var animation in animations)
-                    {
-                        var exportedAnimation = exportedModel.CreateAnimation(animation.Name);
-                        var rotationDict = new Dictionary<string, Dictionary<float, Quaternion>>();
-                        var translationDict = new Dictionary<string, Dictionary<float, Vector3>>();
+        private void LoadEntityMeshes(ModelRoot exportedModel, Scene scene, VEntityLump entityLump)
+        {
+            foreach (var entity in entityLump.GetEntities())
+            {
+                var modelName = entity.GetProperty<string>("model");
+                if (string.IsNullOrEmpty(modelName))
+                {
+                    // Only worrying about models for now
+                    continue;
+                    // TODO: Think about adding lights with KHR_lights_punctual
+                }
+                var className = entity.GetProperty<string>("classname");
+                if (className == "vr_teleport_area" || className == "trigger_multiple")
```

Comment:
Not the biggest fan of doing a filter on entities here, there's gonna be more different triggers too.

Reply:
I'll take that out.